presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -496,6 +506,149 @@ public CounterStat getHighMemorySplitSource()
+                                ValueSet columnPredicateValueSet = domains.get().get(new Subfield(predicateColumnEntry.getKey())).getValues();
+                                if (!columnPredicateValueSet.overlaps(columnsStatsValueSet.get())) {
+                                    pruned = true;
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                if (pruned) {
+                    prunedPartitionNamesBuilder.add(entry.getKey());
+                }
+                else {
+                    partitionBuilder.put(entry.getKey(), entry.getValue().get());
+                }
+            }
+            return new PartitionBatchInfo(partitionBuilder.build(), prunedPartitionNamesBuilder.build());
+        }
+
+        Map<String, Optional<Partition>> batch = metastore.getPartitionsByNames(
+                tableName.getSchemaName(),
+                tableName.getTableName(),
+                Lists.transform(partitionBatch, HivePartition::getPartitionId));
+        ImmutableMap.Builder<String, Partition> partitionBuilder = ImmutableMap.builder();
+        for (Map.Entry<String, Optional<Partition>> entry : batch.entrySet()) {
+            if (!entry.getValue().isPresent()) {
+                throw new PrestoException(HIVE_PARTITION_DROPPED_DURING_QUERY, "Partition no longer exists: " + entry.getKey());
+            }
+            partitionBuilder.put(entry.getKey(), entry.getValue().get());
+        }
+        return new PartitionBatchInfo(partitionBuilder.build());
+    }
+
+    private static Optional<ValueSet> getColumnStatsValueSet(HiveColumnStatistics statistics, HiveType hiveType)
+    {
+        // TODO: Support more types
+        if (hiveType.equals(HIVE_BYTE) || hiveType.equals(HIVE_SHORT) || hiveType.equals(HIVE_INT) || hiveType.equals(HIVE_LONG)) {
+            return getIntegerColumnStatsValueSet(statistics, hiveType);
+        }
+        else {
+            return Optional.empty();
+        }
+    }
+
+    private static Optional<ValueSet> getIntegerColumnStatsValueSet(HiveColumnStatistics statistics, HiveType hiveType)
+    {
+        if (!statistics.getIntegerStatistics().isPresent()) {
+            return Optional.empty();
+        }
+
+        IntegerStatistics hiveColumnStatistics = statistics.getIntegerStatistics().get();
+        Type integerType;
+        if (hiveType.equals(HIVE_BYTE)) {
+            integerType = TINYINT;
+        }
+        else if (hiveType.equals(HIVE_SHORT)) {
+            integerType = SMALLINT;
+        }
+        else if (hiveType.equals(HIVE_INT)) {
+            integerType = INTEGER;
+        }
+        else if (hiveType.equals(HIVE_LONG)) {
+            integerType = BIGINT;
+        }
+        else {
+            throw new IllegalArgumentException(format("Expect integer hive type, but found: %s", hiveType));
+        }
```

Comment:
Let's add simple translation to `HiveTypeTranslator` and add a round-trip test to it.

Reply:
Will add `HiveType -> Type` translation to the existing `TypeTranslator` interface.

Reply:
Just saw `HiveType#getPrimitiveType`, which would achieve what we want. Will just leverage that :)